### PR TITLE
Backup all wallets

### LINKED ISSUE #954: [Bug] Wallets disappear after restoring from a backup
# Bug Report

### Browser

- Browser: Chrome
- Browser Version: -

### Maskbook

- Maskbook Version: 1.10.9
- Installation: Store
- Build Commit: -

### Expected Behavior

All wallets should remain after restoration.

### Actual Behavior

Clipboard - April 8, 2020 7-53 AM

### How To Reproduce

Backup, reinstall, and restore from the backup file.

### Notes

- We should also make sure that the data of plugin "Red Packet" is included in the backup.
- See also #1337.

### PR DESCRIPTION
**Pull request by guanbinrui**
+ [x] Impl generating the wallet backup
+ [x] Impl recovering from the wallet backup
+ [x] Update restore & backup UI about wallet

create_wallet
restore_wallet

```json
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "type": "object",
  "properties": {
    "name": {
      "type": "string"
    },
    "passphrase": {
      "type": "string"
    },
    "address": {
      "type": "string"
    },
    "createdAt": {
      "type": "integer"
    },
    "updatedAt": {
      "type": "integer"
    },
    "mnemonic": {
      "type": "object",
      "properties": {
        "words": {
          "type": "string"
        },
        "parameter": {
          "type": "object",
          "properties": {
            "path": {
              "type": "string"
            },
            "withPassword": {
              "type": "boolean"
            }
          },
          "required": [
            "path",
            "withPassword"
          ]
        }
      },
      "required": [
        "words",
        "parameter"
      ]
    },
    "privateKey": {
      "type": "object",
      "properties": {
        "crv": {
          "type": "string"
        },
        "ext": {
          "type": "boolean"
        },
        "x": {
          "type": "string"
        },
        "y": {
          "type": "string"
        },
        "key_ops": {
          "type": "array",
          "items": [
            {
              "type": "string"
            },
            {
              "type": "string"
            }
          ]
        },
        "kty": {
          "type": "string"
        },
        "d": {
          "type": "string"
        }
      },
      "required": [
        "crv",
        "ext",
        "x",
        "y",
        "key_ops",
        "kty",
        "d"
      ]
    }
  },
  "required": [
    "name",
    "address",
    "createdAt",
    "updatedAt"
  ]
}
```

close #954

### CODE REVIEW

File: src/extension/background-script/WelcomeServices/restoreBackup.ts [hunk 1]
```diff
@@ -48,6 +50,13 @@ export async function restoreBackup(json: object, whoAmI?: ProfileIdentifier) {
                         )
                     }
                 }
+
+                for (const x of data.wallets) {
+                    const record = WalletRecordFromJSONFormat(x)
+                    if (record.type === 'managed') await importNewWallet(record)
+                    // TODO:
+                    // restore exotic wallets
```
Comment (Jack-Works) [inline src/extension/background-script/WelcomeServices/restoreBackup.ts:58]: No need and no way to "restore exotic wallets"

Comment (guanbinrui) [reply src/extension/background-script/WelcomeServices/restoreBackup.ts:58]: Sure, I'm removing it.

File: src/utils/type-transform/BackupFormat/JSON/version-2.ts [hunk 2]
```diff
@@ -65,6 +65,19 @@ export interface BackupJSONFileVersion2 {
         recipientGroups: string[] // Array<GroupIdentifier.toText()>
         foundAt: number // Unix timestamp
     }>
+    wallets: Array<{
```
Comment (Jack-Works) [inline src/utils/type-transform/BackupFormat/JSON/version-2.ts:68]: I think you should leave it as `wallets?: ...` and make a note that this property is new and not all v2 backup has it.

Comment (guanbinrui) [reply src/utils/type-transform/BackupFormat/JSON/version-2.ts:68]: I filled this field up before upgrading the backup, so all v2 backup will got this.

**Pull request merged**